### PR TITLE
[Bug] [dolphinscheduler-api] in the task define list, when edit task, should show pre node list

### LINKED ISSUE #12694: [Bug] [dolphinscheduler-api] This task has pre node, but they are not showing
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

This task has pre node, but they are not showing
image

### What you expected to happen

show pre node

### How to reproduce

1.create task and pre node
2.in the task define list, click edit

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jackfanwan**
backend : in the task define list, when edit task, add return pre node list

closes: #12694

### CODE REVIEW

**Comment by SbloodyS** [comment]
Please use the correct title and use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, just desc). It would not only connect issue to PR but also close issue automatically when PR is be closed. @jackfanwan

**Comment by jackfanwan** [comment]
> Please use the correct title and use keyword fix: #issue_id or close: #issue_id or closes: #issue_id in your PR describe(not title, just desc). It would not only connect issue to PR but also close issue automatically when PR is be closed. @jackfanwan

done

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java [hunk 1]
```diff
@@ -218,6 +218,12 @@ public class TaskDefinition {
      */
     private TaskExecuteType taskExecuteType;
 
+    /**
+     * process task related list
+     */
+    @TableField(exist = false)
+    private List<ProcessTaskRelation> processTaskRelationList;
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskDefinition.java]: I think it's better to use `TaskDefinitionVo` and don't add these fileds in dao entity.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/vo/TaskDefinitionVo.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * @author fanwanlong
+ */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class TaskDefinitionVo {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/vo/TaskDefinitionVo.java]:
```suggestion
public class TaskDefinitionVo extends TaskDefinition {
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 3]
```diff
@@ -1003,7 +1004,15 @@ public Map<String, Object> queryTaskDefinitionDetail(User loginUser, long projec
             logger.error("Task definition does not exist, taskDefinitionCode:{}.", taskCode);
             putMsg(result, Status.TASK_DEFINE_NOT_EXIST, String.valueOf(taskCode));
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java:1005]: Can we return the taskDefinitionVo uniformly?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/vo/TaskDefinitionVo.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public TaskDefinitionVo() {
+    }
+
+    public TaskDefinitionVo(List<ProcessTaskRelation> processTaskRelationList) {
+        this.processTaskRelationList = processTaskRelationList;
+    }
+
+    public static TaskDefinitionVo fromTaskDefinition(TaskDefinition taskDefinition) {
+        TaskDefinitionVo taskDefinitionVo = new TaskDefinitionVo();
+        taskDefinitionVo.setId(taskDefinition.getId());
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/vo/TaskDefinitionVo.java:47]: nip：we can use `BeanUtils.copyProperties` to do it.

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**